pennylane/operation.py

```diff
@@ -2115,6 +2113,7 @@ def prune(self):
         else:
             obs = Tensor(*self.non_identity_obs)
 
+        qml.QueuingManager.remove(self)
```

Comment:
Why?

Reply:
... We could probably not include this line. 

The idea is that the pruned version takes over from the original version I think.